### ISSUE TITLE
build fails with 4.09.0

### ISSUE THREAD

**Issue by andres-erbsen** [smtcoq/smtcoq#50]
Hi,

I am looking into using smtcoq to prove program logic side conditions in bedrock2, in particular with bit vectors and arithmetic. However, it seems that the build is failing on latest ocaml:

```
CAMLC -c versions/standard/mutils_full.mli
File "versions/standard/mutils_full.mli", line 18, characters 16-23:
18 | val numerator : Num.num -> Big_int.big_int
                     ^^^^^^^
Error: Unbound module Num
make[1]: *** [Makefile:595: versions/standard/mutils_full.cmi] Error 2
make: *** [Makefile:327: all] Error 2
[build@archlinux src]$ ocamlc --version
4.09.0
[build@archlinux src]$ coqc --version
The Coq Proof Assistant, version 8.9.0 (October 2019)
compiled on Oct 29 2019 17:55:53 with OCaml 4.09.0
```

Full log: https://gist.github.com/andres-erbsen/ca71cea5cb3655e68ea59528c4b99cab

Is there fundamental reason for why it can't work, or is it just a question of changing some flags somewhere?

Thanks,
Andres

**Comment by ckeller** [comment]
Hi Andres

I tried with the same build (ocaml 4.09.0 and Coq 8.9.0 installed with opam) and did not get any issue :-(

Can you try installing the opam package for SMTCoq instead?
```
opam repo add coq-dev https://coq.inria.fr/opam/extra-dev
opam install coq-smtcoq
```

Best
Chantal

**Comment by andres-erbsen** [comment]
The same error occurs still: https://gist.github.com/andres-erbsen/069a7b60df85f71c9b1c931f0350ed9e#file-gistfile1-txt-L126-L148 . Also, note that both builds posted here were run on fresh virtual machines (but the error also occurs on my laptop).

**Comment by ckeller** [comment]
Hi Andres
It is strange that I cannot reproduce the error but I have tried something which may work: can you try compiling the branch named issue50?
Thank you for your help
Chantal

**Comment by andres-erbsen** [comment]
Now it looks like something is getting double-loaded: [log](https://builds.sr.ht/~andres/job/102470), [script](https://builds.sr.ht/api/jobs/102470/manifest)

**Comment by ckeller** [comment]
Dear Andres
Since you wrote that you prepared a VM, can you share it with me so I can investigate the problem (since I cannot reproduce it on my own machine)? Thank you very much.

**Comment by andres-erbsen** [comment]
https://builds.sr.ht/api/jobs/102470/manifest is the build script that demonstrates this problem on an Arch Linux at sr.ht. I guess it's easiest to use sr.ht to debug this, or just read the script and bring up a similar vm yourself. https://git.sr.ht/~sircmpwn/sr.ht-docs/tree/master/packages.md does link to repositories containing the base image used, but I haven't actually tried downloading it from there and using it. Thanks.

**Comment by ckeller** [comment]
Thank you for pointing this out, it is indeed really useful to debug!
I am investigating why it works on my machine but not on the VM with the same OS and same build, I will let you know soon.

**Comment by ckeller** [comment]
I am not sure what is going on but explicitely switching opam to ocaml-base-compiler.4.09.0 (instead of using the system's ocaml) works:
```
image: archlinux
packages:
- opam
sources:
- https://github.com/smtcoq/smtcoq.git
tasks:
- opam: |
    opam init --no-setup
    opam switch create ocaml-base-compiler.4.09.0
    eval $(opam env)
    opam install -y coq.8.9.0
- smtcoq: |
    eval $(opam env)
    cd smtcoq/src
    ./configure.sh
    make
```

**Comment by gmalecha** [comment]
I haven't looked at this in detail, but the `Num` library is deprecated in favor of `Zarith`. See https://github.com/ocaml/num perhaps there should be a dependency on that package if the compiler is too new.

**Comment by ckeller** [comment]
Thank you for your help. Num is already a dependency of coq and thus installed by opam.

**Comment by andres-erbsen** [comment]
Fixed in #69
